beacon-light-client/plonky2/circuits_executables/src/crud/common.rs

```diff
@@ -200,13 +208,14 @@ pub async fn fetch_final_layer_input(con: &mut Connection) -> Result<FinalCircui
 pub async fn save_balance_proof<const N: usize>(
     con: &mut Connection,
     proof_storage: &mut dyn ProofStorage,
+    protocol: String,
     proof: ProofWithPublicInputs<GoldilocksField, PoseidonGoldilocksConfig, 2>,
     level: u64,
     index: u64,
 ) -> Result<()> {
-    let proof_index = format!(
-        "{}:{}:{}",
-        VALIDATOR_COMMITMENT_CONSTANTS.balance_verification_proof_storage, level, index
+    let proof_key = format!(
+        "{}:{}:{}:{}",
+        VALIDATOR_COMMITMENT_CONSTANTS.balance_verification_proof_storage, level, protocol, index
```

Comment:
I think it's better to prefix everything with the protocol for easier inspection in the explorer. The data is unrelated between different protocols, so we could use the protocol as top level fold and have all the data beneath

Reply:
Although I would still prefer protocol:balance_verification_proofs:level:index

Reply:
Just realized you are using this layout only for proof storage. I would suggest unifying the key for the proof and the proof data

Reply:
You are right I am not sure why I thought it is different